spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -200,4 +201,40 @@ def initialize(options = {})
       expect(NukeablePost.new(:bribe => false).supports_nuke?).to be true
     end
   end
+
+  context "stub_supports" do
+    it "overrides class supports from true to false" do
+      expect(Post.supports?(:archive)).to be true
+
+      stub_supports(Post, :archive, :supported => false, :reason => "reasons")
+
+      expect(Post.supports?(:archive)).to be false
+      expect(Post.unsupported_reason(:archive)).to eq("reasons")
+    end
+
+    it "overrides class supports from false to true" do
+      expect(Post.supports?(:delete)).to be false
+
+      stub_supports(Post, :delete)
+
+      expect(Post.supports?(:delete)).to be true
+    end
+
+    it "overrides instance supports from true to false" do
+      expect(Post.new.supports?(:archive)).to be true
+
+      stub_supports(Post.new, :archive, :supported => false, :reason => "reasons")
```

Comment:
I noticed this test verified instance/instance (i.e. `stub_supports(Post.new)` (on the instance) and then tests `Post.new` (on the instance)).  And the other test verify class/class  Should there also be a test combination for class/instance where someone does `stub_supports(Post)` (on the class), but then checks on Post.new (on the instance).

Reply:
I've never run into this use case. Can't even think of a use case that would require both. Currently, I either test `:create`, which is on the class, or something like `:archive` which is on the instance.

I guess it does remove some knowledge from the developer. We have implemented something similar for our api tests that say "Just give the test user certain privileges" and it doesn't make the developer get too into the weeds.

- This uses `allow` instead of `expect` for `supports?`. I would feel more comfortable changing this to at least 1 time to handle one spec that I wrote. The expect did catch unnecessary stubbing a few times for me. With your suggesting that is removed.
- This also stubs any instance instead of a particular instance, which is a little broad as well. I see us possibly changing this in the future to stub the `Model.find` method like we do in a number of our tests.


I can implement your suggestion with fewer lines than the current implementation, but do you consider this a little broad? Do you want me to move forward with the request. (I am not a fan as you can tell with how many words I wrote)|

Reply:
I'm just asking for a spec.  Honestly, `stub_supports(Post.new)` doesn't even make sense to me, because the `supports` DSL is on the class, so I'd remove that altogether, but I thought maybe I'm just misunderstanding things.

Reply:
ok, I can put a spec in there that shows it doesn't work?

So it is clear. From there we can have a discussion and someone can ask to add it/change the spec to make more sense to everyone (and we possibly add the feature). But adding this as a spec definitely makes it more clear. Which is good.

Reply:
lol. So I added support for this feature and @chessbyte said he didn't like it.

Are we thinking I should roll back the last commit and encourage a person to state that this is a class level supports or an instance level supports?

(I did keep it as a separate commit because I was unsure about it)